### DISCUSSION TITLE
Is there any way of having common templates in helmfiles?

### DISCUSSION THREAD

**Discussion by ennekein** [helmfile/helmfile#1425] [Questions]
Is there any way of having common sections containing state values templating and named templates (using the _define_ template function) in a separate files that can be included in multiple helmfiles and rendered at the same time as them? I'm thinking something like _helpers.tpl_ in Helm.

I've tried the _readFile_ template function, but the result does not get rendered.

I've also tried including the common file in the _bases_ section, but it gets rendered before the main helmfile, so I lose the named templates.

**Comment by yxxhero** [comment]
could you post a example?

**Comment by ennekein** [comment]
Here is something that I'd like to add to all my helmfiles:

```
{{- define "common.registryPrefix" -}}
{{- if .Values.Install.registry.prefix -}}
/{{ .Values.Install.registry.prefix }}
{{- end -}}
{{- end -}}
{{- define "common.registryChartsPath" -}}
{{- if eq .Values.Install.localCharts true -}}
./charts
{{- else -}}
harbor-helm{{ template "common.registryPrefix" . }}
{{- end -}}
{{- end }}
```
